midend/flattenUnions.cpp

```diff
@@ -0,0 +1,359 @@
+/*
+Copyright 2022 Intel Corp.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "flattenUnions.h"
+
+namespace P4 {
+const IR::MethodCallStatement*
+HandleValidityHeaderUnion::processValidityForStr(const IR::Statement *s, const IR::Member *m,
+                                                 cstring headerElement, cstring setValid) {
+    auto method = new IR::Member(s->srcInfo, new IR::Member(m->expr,IR::ID(headerElement)),
+                                 IR::ID(setValid));
+    auto mc = new IR::MethodCallExpression(s->srcInfo, method,
+                                           new IR::Vector<IR::Argument>());
+    typeMap->setType(method, new IR::Type_Method(IR::Type_Void::get(),
```

Comment:
is this necessary? usually we only set types if they are needed right away, otherwise the next typeChecking pass can do it.

Reply:
Removed